File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let task = [];
    
    for(let i = 0; i < priorities.length; i++) {
        for(let j = 0; j < priorities.length; j++) {
            if(Math.max(...priorities) === priorities[j]) {
                task.push(j);
                priorities[j] = 0;
            }
            if(task.length === priorities.length)
                break;
        }
        if(task.length === priorities.length)
                break;
    }
    
    return task.indexOf(location) + 1;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def find(q, priority, target):
    work = 0
    while q:
        p, idx = q.popleft()
        if sum(priority[p+1:]) > 0:
            q.append((p, idx))
        else:
            work += 1
            priority[p] -= 1
            if idx == target:
                break
    return work


def solution(priorities, location):
    q = deque()
    priority = [0] * 10
    
    for i in range(len(priorities)):
        priority[priorities[i]] += 1
        q.append((priorities[i], i))
    
    return find(q, priority, location)